lib/shared/permissions.js

```diff
@@ -101,7 +116,22 @@ exports.getEnvironmentCommandPrefix = (environment) => {
     }
 
     if (isWindows) {
-      return [ 'set', `${key}=${value}`, '&&' ];
+      const definition = `${key}=${value}`;
+      return [
+        'set',
+        _.attempt(() => {
+          if (/\s/.test(definition)) {
+            return quoteString(definition);
+          }
+
+          return definition;
+        }),
+        '&&'
+      ];
+    }
+
+    if (/\s/.test(value)) {
```

Comment:
Forgot to say earlier, but when I had a look at `command-join` it was doing all the escaping we needed inside a for loop, and not inside a separate function that we could re-use here :(